packages/appium/lib/appium.js

```diff
@@ -327,18 +374,31 @@ class AppiumDriver extends DriverCore {
 
       const driverInstance = /** @type {ExternalDriver} */ (new InnerDriver(this.args, true));
 
-      // We want to assign security values directly on the driver. The driver
-      // should not read security values from `this.opts` because those values
-      // could have been set by a malicious user via capabilities, whereas we
-      // want a guarantee the values were set by the appium server admin
-      if (this.args.relaxedSecurityEnabled) {
+      // If anything in the umbrella driver's insecure feature configuration applies to this driver,
+      // assign it to the driver instance
+      if (this.relaxedSecurityEnabled) {
         this.log.info(
-          `Applying relaxed security to '${InnerDriver.name}' as per ` +
-            `server command line argument. All insecure features will be ` +
-            `enabled unless explicitly disabled by --deny-insecure`,
+          `Enabling relaxed security for this session as per the server configuration. ` +
+            `All insecure features will be enabled unless explicitly disabled by --deny-insecure`,
         );
         driverInstance.relaxedSecurityEnabled = true;
       }
+      const allowedDriverFeatures = filterInsecureFeatures(this.allowInsecure, driverName);
```

Comment:
maybe it would make sense to extract this block into a separate method to improve readability